### DISCUSSION TITLE
Overriding global sass variables

### DISCUSSION THREAD

**Discussion by FrenchGithubUser** [quasarframework/quasar#14347] [CLI - Capacitor / Cordova modes] [answered]
Hi, I am building a mobile app with a header and a footer. To display pages correctly, I created a variable in quasar.variables.scss `$page-height` (so the content doesn't stay hidden under the footer for example)

The problem is the following : when the keyboard appears on screen, I hide the footer with `v-show`, but the `$page-height` variable remains the same and causes some visual problems (useless bottom white space). 

So I was wondering if it was possible to override this global variable when the keyboard is on screen.

Thanks in advance !

**Comment by FrenchGithubUser** [comment, marked as answer]
I ended up using css variables and changing them with `document.documentElement.style.setProperty("--bottom-nav-bar-height", "65px")`

**Comment by yusufkandemir** [comment]
If we let aside SASS, CSS, etc. variables, you can handle it via `styleFn` prop of `QPage`. https://quasar.dev/layout/page#style-fn

If you made it work and is easier to modify it using SASS/CSS variables, then you can use them as well. SASS variables are static, so you can't update them on build time. You would use CSS variables for that purpose. I would recommend using [`setCssVar`](https://quasar.dev/style/color-palette#util-setcssvar) for convenience though. It assumes the variables have a `q-` prefix, but it's easier to work with.
